### PR TITLE
Test/stable/fact 1614/update verify fact tests and move to facts directory

### PR DESCRIPTION
**Pull request by ericldelaney**
(FACT-1614) Move tests into facts directory and update

Moved the fact checking tests into the facts directory
Updated for code style, TestRail ID, and risk tags
Removed tests/facts/verify_facts_match_schema.rb as its covered by unit tests

### CODE REVIEW

**Comment by ericldelaney** [comment]
In the first commit I only moved the files so if you select the 	5741d31 commit its much easier to see what got changed.

Tested on:
arista4-32a, ciscoxr-64a%7Bhypervisor=vmpooler%7D, cisconx-64a%7Bhypervisor=vmpooler%7D, cumulus25-64a, debian7-64a, debian7-32a, debian8-64a, debian8-32a, fedora24-64a, fedora25-64a, osx1010-64a, osx1011-64a, osx1012-64a, redhat5-64a, redhat6-64a, redhat6-32a, redhat7-64a, sles11-64a, sles11-32a, sles12-64a, solaris10-64a, solaris11-64a, ubuntu1404-64a, ubuntu1404-32a, ubuntu1604-64a, ubuntu1604-32a, ubuntu1610-64a, ubuntu1610-32a, windows10ent-64a, windows10ent-32a, windows2008r2-64a, windows2008r2-6432a, windows2012r2-64a, windows2012r2-6432a, windows2016-64a, windows2016-6432a

File: acceptance/tests/facts/verify_facts_match_schema.rb [hunk 1]
```diff
@@ -0,0 +1,111 @@
+test_name 'C62454: verify facts return from facter match the schema' do
```
Comment (johnduarte) [inline acceptance/tests/facts/verify_facts_match_schema.rb]: I would argue that this is a set of unit tests that should be covered at that layer and do not belong in an acceptance test.

File: acceptance/tests/facts/verify_tmpfs_file_system.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    create_remote_file(agent, manifest, manifest_content)
+
+    teardown do
+      on(agent, "umount #{mount_point}")
+      on(agent, "rm -rf #{mount_point} #{manifest_dir}")
+    end
+
+    step "Apply the manifest to mount directory '#{mount_point}'" do
+      on(agent, puppet("apply #{manifest}"), :acceptable_exit_codes => [0,2]) do |puppet_apply|
+        assert_no_match(/Error/, puppet_apply.stdout, 'Unexpected error was detected!')
```
Comment (johnduarte) [inline acceptance/tests/facts/verify_tmpfs_file_system.rb]: Should this also have an assertion that no `Error` is present in STDOUT?

File: acceptance/tests/facts/verify_facts_match_schema.rb (hunk 1 above)
Comment (kurtwall) [inline acceptance/tests/facts/verify_facts_match_schema.rb]: +1. I believe it already is covered in facter/lib/tests/facts/schema.cc.

File: acceptance/tests/facts/verify_tmpfs_file_system.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# This test is intended to demonstrate that mount resource can mount tmpfs file systems
+# and the mount facter mountpoints should show the mount as tmpfs
+test_name 'C98163: mountpoints fact should show mounts on tmpfs' do
+  tag 'risk:high'
+
+  require 'facter/acceptance/user_fact_utils'
+  extend Facter::Acceptance::UserFactUtils
+
+  confine :except, :platform => 'windows'
```
Comment (kurtwall) [inline acceptance/tests/facts/verify_tmpfs_file_system.rb]: So many `confine :except`. Time to make this `confine, :platform => blah` instead?

File: acceptance/tests/facts/verify_tmpfs_file_system.rb (hunk 2 above)
Comment (ericldelaney) [inline acceptance/tests/facts/verify_tmpfs_file_system.rb]: Do you mean stderr? (Its already testing stdout)

File: acceptance/tests/facts/verify_tmpfs_file_system.rb (hunk 3 above)
Comment (ericldelaney) [inline acceptance/tests/facts/verify_tmpfs_file_system.rb]: We could, but if we add a new platform that supports tmpfs, it won't get tested.

File: acceptance/tests/facts/verify_tmpfs_file_system.rb (hunk 2 above)
Comment (johnduarte) [inline acceptance/tests/facts/verify_tmpfs_file_system.rb]: :man_facepalming: yes, I meant STDERR. Thanks.

Comment (ericldelaney) [inline acceptance/tests/facts/verify_tmpfs_file_system.rb]: ok sure.

File: acceptance/tests/facts/verify_facts_match_schema.rb (hunk 1 above)
Comment (ericldelaney) [inline acceptance/tests/facts/verify_facts_match_schema.rb]: I haven't started looking at unit tests yet. Its next on my list after I finish updating the last 13 acceptance tests.

After looking at the Unit test, yeah we can probably lose this test at the acceptance level. I'll remove it.

Comment (kurtwall) [inline acceptance/tests/facts/verify_facts_match_schema.rb]: 👍 That's cool. It rang a bell as something I'd seen.

File: acceptance/tests/facts/verify_tmpfs_file_system.rb (hunk 3 above)
Comment (ericldelaney) [inline acceptance/tests/facts/verify_tmpfs_file_system.rb]: I checked you'd trade 7 excludes for 6 includes.

**Comment by ericldelaney** [comment]
@johnduarte @kurtwall Updated

**Comment by kurtwall** [comment]
I'll merge when it passes travis and appveyor.

**Pull request merged**